sys/phydat/phydat_str.c

```diff
@@ -18,149 +18,229 @@
  * @}
  */
 
-#include <stdio.h>
+#include <errno.h>
 #include <stdint.h>
+#include <stdio.h>
 
 #include "assert.h"
+#include "flash_utils.h"
 #include "fmt.h"
 #include "phydat.h"
 
 void phydat_dump(phydat_t *data, uint8_t dim)
 {
     if (data == NULL || dim > PHYDAT_DIM) {
-        puts("Unable to display data object");
+        flash_puts(TO_FLASH("Unable to display data object"));
         return;
     }
-    printf("Data:");
+    flash_printf(TO_FLASH("Data:"));
 
     if (data->unit == UNIT_TIME) {
         assert(dim == 3);
-        printf("\t%02d:%02d:%02d\n",
-               data->val[2], data->val[1], data->val[0]);
+        flash_printf(TO_FLASH("\t%02d:%02d:%02d\n"),
```

Comment:
Could we enforce the first argument of `flash_printf` to always be in the flash? (so the `TO_FLASH` wrapper is not needed to be manually added at every call site)

Reply:
How about

```C
#define flash_str_printf(s, ...) flash_printf(TO_FLASH(s), __VA_ARGS__)
```